common/djangoapps/student/management/tests/test_bulk_unenroll.py

```diff
@@ -0,0 +1,105 @@
+from tempfile import NamedTemporaryFile
+
+from django.core.management import call_command
+from xmodule.modulestore.tests.django_utils import SharedModuleStoreTestCase
+from xmodule.modulestore.tests.factories import CourseFactory
+from testfixtures import LogCapture
+
+from course_modes.tests.factories import CourseModeFactory
+from student.tests.factories import UserFactory
+from student.models import CourseEnrollment, User
+
+
+LOGGER_NAME = 'student.management.commands.bulk_unenroll'
+
+
+class BulkUnenrollTests(SharedModuleStoreTestCase):
+    """Test Bulk un-enroll command works fine for all test cases."""
+    def setUp(self):
+        super(BulkUnenrollTests, self).setUp()
+        self.course = CourseFactory.create()
+        self.audit_mode = CourseModeFactory.create(
+            course_id=self.course.id,
+            mode_slug='audit',
+            mode_display_name='Audit',
+        )
+
+        self.user_info = [
+            ('amy', 'amy@pond.com', 'password'),
+            ('rory', 'rory@theroman.com', 'password'),
+            ('river', 'river@song.com', 'password')
+        ]
+        self.enrollments = []
+        self.users = []
+
+        for username, email, password in self.user_info:
+            user = UserFactory.create(username=username, email=email, password=password)
+            self.users.append(user)
+            self.enrollments.append(CourseEnrollment.enroll(user, self.course.id, mode='audit'))
+
+    def test_user_not_exist(self):
+        """Verify that warning user not exist is logged for non existing user."""
+        with NamedTemporaryFile() as csv:
+            csv.write("user_id,username,email,course_id\n")
+            csv.writelines("111,test,test@example.com,course-v1:edX+DemoX+Demo_Course\n")
+            csv.seek(0)
+
+            with LogCapture(LOGGER_NAME) as log:
+                call_command("bulk_unenroll", "--csv_path={}".format(csv.name))
+                log.check(
+                    (
+                        LOGGER_NAME,
+                        'WARNING',
+                        'User with username {} or email {} does not exist'.format('test', 'test@example.com')
+                    )
+                )
+
+    def test_invalid_course_key(self):
+        """Verify in case of invalid course key warning is logged."""
+        with NamedTemporaryFile() as csv:
+            csv.write("user_id,username,email,course_id\n")
```

Comment:
Move these three lines to a helper like `_write_test_csv(csv, lines)`.  The helper could loop over the lines and output them.  It could even add the `\n` at the end if it isn't there, so that the test will look cleaner and it would be less prone to errors.

Reply:
I had imagined lines would be a list of strings, but it is fine as-is.  Thanks for the move.